src/auth.service.ts

```diff
@@ -38,11 +38,11 @@ export default class AuthService {
 	}
 
 	async auth(auth: Auth): Promise<object> {
-		const user = await this.userService.findByEmail(auth.email);
+		const user = await this.userService.findByEmailWithPassword(auth.email);
 
 		if (!user) throw new BadRequestException('Invalid credentials');
```

Comment:
This might be better if you used `UnauthorizedException`?
https://docs.nestjs.com/exception-filters

Reply:
Or ForbiddenException

Reply:
User error = 400 bad response. Want to keep unauthorised and forbidden for their exclusive use only.